### DISCUSSION TITLE
Fetching External Data Sources From Custom Cosmos Module

### DISCUSSION THREAD

**Discussion by AntoninPS** [cosmos/community#40] [Chain Builders (Cosmos SDK)] [answered]
I am building a custom Cosmos module intended to be integrated into the Cosmos Hub chain. 

The module needs to fetch data from external data sources, such as CoinMarketCap for example. 

What are the best practices to fetch data from external sources from my module's keeper methods?

1. HTTP API
I tried making simple http calls to the API using the Golang net/http package. This works as expected, however to access the data sources I need an API key. Where can I privately store the API key? 
2. Oracle Integration
I also looked into Oracle integration but found very little documentation / discussions about it. I looked into Band Protocol but i'm having difficulties understanding how to implement it in my module. 

Any help or links to helpful resources would be appreciated :slight_smile:

Thanks

**Comment by tac0turtle** [comment, marked as answer]
Hey there is a new oracle module that the skip team has been working on. Its not open source yet but may be soon. Until then ojo network is a good example of an oracle network (https://ojo.network/)
